WordPress/src/main/java/org/wordpress/android/widgets/WPLoginInputRow.java

```diff
@@ -0,0 +1,236 @@
+package org.wordpress.android.widgets;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.os.Build;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.Nullable;
+import android.support.design.widget.TextInputLayout;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.view.KeyEvent;
+import android.view.inputmethod.EditorInfo;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import org.wordpress.android.R;
+import org.wordpress.android.util.ViewUtils;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Compound view composed of an icon and an EditText
+ */
+public class WPLoginInputRow extends RelativeLayout {
+
+    public interface OnEditorCommitListener {
+        void OnEditorCommit();
+    }
+
+    private ImageView mIcon;
+    private TextInputLayout mTextInputLayout;
+    private EditText mEditText;
+
+    private List<Integer> mNewIds;
+
+    public ImageView getIcon() {
+        return mIcon;
+    }
+
+    public EditText getEditText() {
+        return mEditText;
+    }
+
+    public WPLoginInputRow(Context context) {
+        super(context);
+        init(context, null);
+    }
+
+    public WPLoginInputRow(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        init(context, attrs);
+    }
+
+    public WPLoginInputRow(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        init(context, attrs);
+    }
+
+    private void init(Context context, AttributeSet attrs) {
+        inflate(context, R.layout.login_input_row, this);
+
+        mIcon = (ImageView) findViewById(R.id.icon);
+        mTextInputLayout = (TextInputLayout) findViewById(R.id.input_layout);
+        mEditText = (EditText) findViewById(R.id.input);
+
+        if (attrs != null) {
+            TypedArray a = context.getTheme().obtainStyledAttributes(attrs, R.styleable.wpLoginInputRow, 0, 0);
```

Comment:
I intentionally left out the `else`s so the default attributes of the components are applied anyway. I'd suggest we introduce `else`s on a case-by-case manner when needed.

By the way, in a different PR (https://github.com/wordpress-mobile/WordPress-Android/pull/6341#issuecomment-316314210) I did introduce an `else` case for the icon since we specifically want some behaviour to happen on the `else`.